### PR TITLE
Fixing roundtrip from qcschema to qcschema

### PR DESCRIPTION
**Pull request by pavankum**
`Molecule.from_qcschema` will now check for cmiles entries in extras too, `Molecule.to_qcschema` will write ['canonical_isomeric_explicit_hydrogen_mapped_smiles'] in extras dict, one test included to check the roundtrip. Fix for issue #720. 

- [x] Tag issue being addressed
- [x] Add [tests](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/tests)
- [x] Update docstrings/[documentation](https://github.com/openforcefield/openforcefield/tree/master/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [x] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

### CODE REVIEW

**Comment by dotsdl** [comment]
Reviewing!

**Comment by dotsdl** [comment]
@pavankum pushed changes to `from_qcschema` based on current understanding. Designed to also give us a conformer for resulting openff `Molecule` from `geometry` field, which was missing in original implementation.

**Comment by dotsdl** [comment]
Also updated docstrings to conform to QC* terminology as closely as possible.

**Review by dotsdl — COMMENTED** [review]
Thank you for spearheading this @pavankum! In particular, thank you for improving the test coverage of these functions, and for improving the CMILES logic to handle the case of molecule records with `extras` instead of `attributes` populated.

**Review by dotsdl — APPROVED** [review]

**Comment by dotsdl** [comment]
@pavankum I can resolve conflicts with `master` if you like?

**Comment by pavankum** [comment]
@dotsdl Thank you very much for reviewing and applying a better fix for this issue :)

**Comment by dotsdl** [comment]
@j-wags if you have no objections, please merge!

File: openff/toolkit/tests/test_molecule.py [hunk 1]
```diff
@@ -1799,10 +1799,10 @@ def assert_check():
...
         assert_check()
         assert qcschema.extras["test_tag"] == "test"
-        # now run again when no extras
-        qcschema = ethanol.to_qcschema()
-        assert_check()
-        assert qcschema.extras is None
+        # # now run again when no extras
+        # qcschema = ethanol.to_qcschema()
+        # assert_check()
+        # assert qcschema.extras is None
```
Comment (j-wags) [inline openff/toolkit/tests/test_molecule.py:1805]: @pavankum Could you take a pass at this and either delete it, or change it to test some important part of the new functionality?

Comment (pavankum) [reply openff/toolkit/tests/test_molecule.py]: resolved by modifying old test to check for canonical smiles in extras entries

**Pull request merged**